### ISSUE TITLE
Polymorphic Function Error Message

### ISSUE THREAD

**Issue by schackbrian2012** [racket/typed-racket#610]
### What version of Racket are you using?
6.10.1

### What program did you run?
`(string (car (cons 0 1)))`

### What should have happened?
I expected to get an error message, but not this error message. "car could not be applied to arguments." car has only one argument singular not arguments plural. And the error sounds like the argument to car is wrong, but that argument is correct. The actual error is the output type of car doesn't match the input type of string. Also, the full type for car in the error message is difficult to read –– it seems like too much information.

### If you got an error message, please include it here.
```
Type Checker: Polymorphic function `car' could not be applied to arguments:
Types: (Pairof a b) #f * -> (a : ((! (car (0 0)) False) | (: (car (0 0)) False)) : (car (0 0)))
       (Listof a) #f * -> a
Arguments: (Pairof Zero One) #f *
Expected result: Char
```

**Comment by samth** [comment]
I think what we want to do here is change "arguments" to "argument" and add "with the expected result type" in the message.

**Comment by schackbrian2012** [comment]
@samth Yes, that is correct. Also, the type for cons is hard to read. It includes characters that I couldn't find in the typed/racket documentation like |.
